src/items/items.controller.ts

```diff
@@ -2,71 +2,67 @@ import {
-import { ApiTags } from '@nestjs/swagger';
+import { CreateItemDto } from './dto/create-item.dto';
+import { UpdateItemDto } from './dto/update-item.dto';
 import { ApiItems } from './items.swagger';
+import { PaginationQueryDto } from './dto/pagination-query.dto';
+import { PositiveIntPipe } from 'src/common/pipes/positive-int/positive-int.pipe';
+import { CategoryQueryDto } from './dto/category-query.dto';
 
-@ApiTags('items')
-@Controller('items') // '/items'경로 요청 처리
+@Controller('items')
 export class ItemsController {
   constructor(private readonly itemsService: ItemsService) {}
 
-  // GET /items 요청 처리
-  @Get()
-  @ApiItems.getAllItems()
-  async getAllItems() {
-    return await this.itemsService.getAllItems(); //service에서 아이템 목록을 받아서, 반환(return)
+  //1. 새로운 아이템 추가 : POST /items
+  @Post()
+  @ApiItems.createItem()
+  @HttpCode(201)
+  async createItem(@Body() createItemDto: CreateItemDto) {
+    return await this.itemsService.createItem(createItemDto);
   }
 
-  // POST /items/buy 요청 처리
-  @Post('buy')
-  @HttpCode(204)
-  @ApiItems.buyItem()
-  async buyItem(@Body() buyItemDto: BuyItemDto): Promise<void> {
-    await this.itemsService.buyItem(buyItemDto);
-  }
-
-  // GET /items/user/:userId 요청 처리
-  @Get('users/:userId')
-  @ApiItems.getUserItems()
-  async getUserItems(@Param('userId') userId: number) {
-    return await this.itemsService.getUserItems(userId);
+  //2. 전체 아이템 목록 조회 : GET /items --> pagination 적용
+  @Get()
+  @ApiItems.getAllItems()
+  @HttpCode(200)
+  async getAllItems(@Query() paginationQuery: PaginationQueryDto) {
+    return this.itemsService.getAllItems(paginationQuery);
   }
 
-  // POST /items/equip 요청 처리
-  @Post('equip')
-  @HttpCode(204)
-  @ApiItems.equipItem()
-  async equipItem(@Body() equipItemDto: EquipItemDto) {
-    await this.itemsService.equipItem(equipItemDto);
+  //4. 카테고리별 아이템 조회 : GET /items/category?mainCategoryId=1&subCategoryId=2
+  @Get('category')
+  @ApiItems.getItemsByCategory()
+  @HttpCode(200)
+  async getItemsByCategory(@Query() query: CategoryQueryDto) {
+    return this.itemsService.getItemsByCategory(
+      query.mainCategoryId,
+      query.subCategoryId,
+    );
   }
 
-  // POST /items/unequip 요청 처리
-  @Post('unequip')
-  @HttpCode(204)
-  @ApiItems.unequipItem()
-  async unequipItem(@Body() unequipItemDto: UnequipItemDto) {
-    await this.itemsService.unequipItem(unequipItemDto);
+  //3. 단일 아이템 조회 : GET /items/:id
+  @Get(':id')
+  @ApiItems.getItemById()
+  @HttpCode(200)
+  async getItemById(@Param('id', PositiveIntPipe) id: number) {
+    return this.itemsService.getItemById(id);
   }
 
-  // DELETE /items/:userId/:itemId 요청 처리
-  @Delete('users/:userId/:itemId')
-  @HttpCode(204)
-  @ApiItems.deleteUserItem()
-  async deleteUserItem(
-    @Param('userId') userId: number,
-    @Param('itemId') itemId: number,
+  //5. 아이템 수정 : PATCH /items/:id
+  @Patch(':id')
+  @ApiItems.updateItem()
+  @HttpCode(200)
+  async updateItem(
+    @Param('id') id: number,
```

Comment:
```suggestion
    @Param('id', PositiveIntPipe) id: number,
```
PositiveIntPipe가 빠져있어서 현재 id는 string 타입이네요.

Reply:
감사합니당!!!!!